### PR TITLE
Added keyboard shortcuts for ghc-mod in haskell.

### PR DESCRIPTION
**Pull request by Tritlo**
I added some keyboard shortcuts for ghc-mod, to use in haskell-mode, since there were none previously. These include shortcuts to get info, type, and (the one which prompted this).

### CODE REVIEW

**Comment by d12frosted** [comment]
Thank you for your PR. Indeed Spacemacs lacks better coverage for `ghc-mod` functions.

The only thing I would like to notice is that bindings for get info and get type functions are set [here](https://github.com/syl20bnr/spacemacs/blob/develop/layers/+lang/haskell/packages.el#L135). This is done in such way due the fact that different 'extensions' to `haskell-mode` set this binding to the appropriate function, and we don't want to track which function should be used. Also Spacemacs has a convention about bindings for getting type of symbol - it should be `m h t`. So your commit kinda breaks it.

**Comment by joehillen** [comment]
Should probably also add these bindings to the README.

**Comment by robbyoconnor** [comment]
If you add keybindings, **ALWAYS, ALWAY, ALWAYS** document them!

**Comment by d12frosted** [comment]
> Should probably also add these bindings to the README.

Yes, good call.

**Comment by Tritlo** [comment]
Alright, I've fixed your grievances, and squashed the fixes together. I removed some redundancies and added documentation. Will you please review again?

File: layers/+lang/haskell/packages.el [hunk 1]
```diff
@@ -45,10 +45,25 @@
     :if haskell-enable-ghc-mod-support
     :init (add-hook 'haskell-mode-hook 'ghc-init)
     :config
-    (when (configuration-layer/package-usedp 'flycheck)
-      ;; remove overlays from ghc-check.el if flycheck is enabled
-      (set-face-attribute 'ghc-face-error nil :underline nil)
-      (set-face-attribute 'ghc-face-warn nil :underline nil))))
+    (progn
+      (if (fboundp 'which-key-declare-prefixes)
+          (which-key-declare-prefixes
```
Comment (d12frosted) [inline layers/+lang/haskell/packages.el]: Better to use `(spacemacs/declare-prefix-for-mode 'haskell-mode "mm" "haskell/ghc-mod")`.

**Comment by d12frosted** [comment]
@Tritlo Thanks for the update! Sorry that I haven't noticed that thing with `prefix` before.

**Comment by Tritlo** [comment]
Alright, added the prefix, and updated the documentation a bit to be more in line with the other documentation.

**Comment by d12frosted** [comment]
Cool, @TheBB  summoning you :smile:

**Comment by TheBB** [comment]
Thank you! Cherry-picked in develop. You can safely delete your branch.

**Pull request closed without merging**